cis194/week6/daniel-deng/src/Exercise05.hs

```diff
@@ -0,0 +1,13 @@
+module CIS194.Homework06.Exercise05 (nats, ruler) where
+
+import CIS194.Homework06.Exercise03 (Stream(..))
+import CIS194.Homework06.Exercise04 (streamRepeat, streamMap, streamFromSeed)
+
+nats :: Stream Integer
+nats = streamFromSeed (+1) 0
+
+interleaveStreams :: Stream a -> Stream a -> Stream a
+interleaveStreams (Stream x xs) ys = Stream x . interleaveStreams ys $ xs
+
+ruler :: Stream Integer
+ruler = interleaveStreams (streamRepeat 0) (streamMap (+1) ruler)
```

Comment:
@shineli1984 I find the original (red line) version reads better as it clearly states `interleaveStreams` is of arity 2. But you've explained the rule of less visual distraction. It applies here too? 

```suggestion
ruler = interleaveStreams (streamRepeat 0) . streamMap (+1) $ ruler
```

Reply:
Yes.